ak/ak_status.py

```diff
@@ -0,0 +1,202 @@
+        repo_key = "./%s/%s" % (VENDOR_FOLDER, key)
+    return repo_key
+
+
+@Ak.subcommand("status")
+class AkAnalyse(AkSub):
+    "dependencies status for odoo"
+
+    config = cli.SwitchAttr(
+        ["c", "config"], default=SPEC_YAML, help="Config file", group="IO"
+    )
+
+    def _ensure_viable_installation(self, config):
+        if not local.path(config).is_file():
+            raise Exception("Config file not found.")
+
+    def main(self, *args):
+        config_file = self.config
+        self._ensure_viable_installation(config_file)
+
+        spec_data = yaml.safe_load(open(self.config).read()) or {}
+        serie = False
+        for repo, data in spec_data.items():
+            logger.debug("********** in", repo)
+            remote = False
+            remote_main_branch = False
+            if "src" in data:
+                remote = "origin"
+                serie_candidate = data["src"].split(" ")[-1]
+                remote_main_branch = serie_candidate
+                if (
+                    serie_candidate.endswith(".0")
+                    and serie_candidate.split(".")[0].isdigit()
+                ):
+                    serie = serie_candidate
+
+            elif "merges" in data:
+                for index, merge in enumerate(data["merges"]):
+                    if index == 0:
+                        remote = merge.split(" ")[0]
+                        serie_candidate = merge.split(" ")[-1]
+                        remote_main_branch = serie_candidate
+                        if (
+                            serie_candidate.endswith(".0")
+                            and serie_candidate.split(".")[0].isdigit()
+                        ):
+                            serie = serie_candidate
+            if not serie:
+                raise RuntimeError(f"Unable to figure out Odoo serie for {repo}")
+
+            repo_path = get_repo_key_from_spec(repo)
+            if not local.path(repo_path).is_dir():
+                print(f"{repo_path} NOT  FOUND!")
+                continue
+
+            with local.cwd(repo_path):
+                local_module_versions = {}
+                # fisrt we get the current module versions from their manifest
+                for module in data.get("modules", []):
+                    if local.path(f"{module}/__manifest__.py").is_file():
+                        module_path = f"{module}/__manifest__.py"
+                    elif local.path("__manifest__.py").is_file():
+                        module_path = "__manifest__.py"
+                    else:
+                        continue
+                    version = ((cat < module_path) | grep[f"{serie}."])().split(":")[1]
+                    local_module_versions[module] = (
+                        version.replace('"', "").replace(",", "").strip()
+                    )
+
+                # TODO see if we cannot hide the stdout here:
+                local["git"]["fetch", remote, remote_main_branch] & FG
```

Comment:
to hide from stdout; try:
```suggestion
                local["git"]["fetch", remote, remote_main_branch]()
```

Reply:
the optimistic "no freeze" approach can also work for large projects before they are in production.

But note that changes such as the ones I warned about in the account_payment_order module 6 months ago would have easily broken any small project. So may be we could still freeze many projects and simply filter in ak status to report only major changes so the human check is faster.

As for detecting changes between "builds" we might also save the analysis from the click-odoo update command. But as it is md5 based it is much less informative.

As for patched modules: well eventually we could still report any modules version changed in the git log along with changes in the patch branches. it's not really about trusting a version number but t might still be a better than nothing to know some patched module got a major or minor upstream change, or even a patch level change then if we want to be more cautious.

Reply:
Also, we can check with partner_modules in our instance if some of our other projects are already in a version. It can reduce the cost of each one of us looking all the pr.